src/openfermion/utils/_sparse_tools.py

```diff
@@ -203,6 +203,64 @@ def qubit_operator_sparse(qubit_operator, n_qubits=None):
     sparse_operator.eliminate_zeros()
     return sparse_operator
 
+def get_linear_qubit_operator(qubit_operator, n_qubits=None):
+    """ Return a linear operator with matvec defined to avoid instantiating a
+    huge matrix, which requires lots of memory.
+
+    Args:
+      qubit_operator(QubitOperator): A qubit operator to be applied on vectors.
+
+    Returns:
+      linear_operator(LinearOperator): The equivalent operator which is well
+      defined when applying to a vector.
+
+    """
+    if n_qubits is None:
+        n_qubits = count_qubits(qubit_operator)
+    if n_qubits < count_qubits(qubit_operator):
+        raise ValueError('Invalid number of qubits specified.')
+
+    n_hilbert = 2 ** n_qubits
+
+    def matvec(vec):
+        """matvec for the LinearOperator class."""
+        if len(vec) != n_hilbert:
+            raise ValueError('Invalid length of vector specified: %d != %d'
+                             %(len(vec), n_hilbert))
+
+        retvec = numpy.zeros(vec.shape, dtype=complex)
+        # Loop through the terms.
+        for qubit_term in qubit_operator.terms:
+            vecs = [vec]
+            tensor_factor = 0
+            coefficient = qubit_operator.terms[qubit_term]
+
+            for pauli_operator in qubit_term:
+                # Split vector by half and half for each bit.
+                num_splits = 2 ** (pauli_operator[0] - tensor_factor)
+                if num_splits >= 2:
+                    vecs = [v for iter_v in vecs
```

Comment:
I know what you're doing here, but if you could add a few more lines of comment on your general algorithmic approach to implementing the action of the operators so it's easier to maintain in the future, it would be very helpful.  If there was any reference you used that might suffice, or if you derived it from scratch, just describe a bit the logic behind the approach?

Reply:
Yes, it's from scratch. Comments are added, thanks for the suggestion!